### ISSUE TITLE
[BottomNavigationBar] Initiate tabs programatically doesn't work

### ISSUE THREAD

**Issue by alexandruantonica** [Akylas/nativescript-material-components#84]
@henrychavez Something seems to be wrong [here](https://github.com/Akylas/nativescript-material-components/blob/fb43a4d793f74ea0f1d5680f118de10cf1beffda/src/bottomnavigationbar/bottomnavigationbar-common.ts#L192). The construction is not available anymore in `BottomNavigationTabBase`.

This is the reason why I get the error "Expected 0 arguments, but got 1". 

Is there a good reason why you removed the constructor from this version. In `nativescript-bottom-navigation` seems to be and works fine.

**Comment by farfromrefug** [comment]
My bad will fix it

**Comment by alexandruantonica** [comment]
@farfromrefug It's a Typescript error that doesn't allow me to add more parameters to the constructor. 

That constructor is necessary if you want to initiate the BottomNavigationBar like that 
```
    <BottomNavigationBar
        [tabs]="tabs"
        activeColor="red"
        inactiveColor="yellow"
        backgroundColor="black"
        (tabSelected)="onBottomNavigationTabSelected($event)"
        (tabPressed)="onBottomNavigationTabPressed($event)"
        row="1"
    ></BottomNavigationBar>
```
This is one way the documentation recommend. [Check it out here](https://github.com/Akylas/nativescript-material-components/tree/master/packages/nativescript-material-bottomnavigationbar#angular)
```
ngOnInit() {
    this.bottomBar.structure = [
            {
                title: this.translates.instant("HOMEPAGE.TITLE"),
                icon: "res://icon_home_bb",
                pointer: "homepage",
                selectable: true
            },
            {
                title: this.translates.instant("CONTACTS.TITLE"),
                icon: "res://icon_contacts_bb",
                pointer: "contacts",
                selectable: false
            }
        ];

   this.tabs = this.bottomBar.structure.map(tab => {
            return new BottomNavigationTab({
                title: tab.title,
                icon:  tab.icon
                isSelectable: tab.selectable
            });
        })
}
```

**Comment by farfromrefug** [comment]
Ok got it will bring it back in a future version.]
In the meantime you can use this instead:
```ts
this.tabs = this.bottomBar.structure.map(tab => {
            const nTab = new BottomNavigationTab();
            nTab.title = tab.title;
            nTab.icon = tab.icon;
            nTab.isSelectable = tab.isSelectable;
           return nTab;
        })
```
That s actually the way it should be done with {N}

**Comment by alexandruantonica** [comment]
@farfromrefug Cool. Thanks
It would be nice to add that in documentation as well.

**Comment by farfromrefug** [comment]
Yes that will need to be done. Both way will work though

**Comment by farfromrefug** [comment]
@alexandruantonica released the fix in a beta version 3.0.0. Be careful though it requires ```@nativescript```
